### DISCUSSION TITLE
The auto-publishing of translations

### DISCUSSION THREAD

**Discussion by krieghan** [wagtail/wagtail-localize#593] [Ideas]
In https://github.com/wagtail/wagtail-localize/issues/553, https://github.com/wagtail/wagtail-localize/pull/511, and finally https://github.com/wagtail/wagtail-localize/commit/a7b009939d3ee4af3fc1127588f3aa827f874b82, a decision seems to have been made that it is reasonable to assume that the publish state of a translation should be the same as the publish state of the page that the translation is for - that is to say, if a page is in draft and a translation is added, it should be in draft, and likewise if a page is already published and a translation is added, it should be automatically published.

I'm wondering if this guiding assumption is reasonable.  From my (admittedly naive) perspective, it seems as if a translation should be set to draft when it is added, and published manually, whether its root page is live or in draft.  Is there something I'm missing?

**Comment by zerolab** [comment]
Hey @krieghan,

This is one of the more common patterns, and is something our projects and clients need.
In lots of cases, you get the translated version, if it exists, or the original language. This ensures continuity. 

e.g if you have two pages, Page B a child of and linking to Page A, both published. You go and translate Page B, but not A. This would carry over the tree setup, but you only translate B. What happens with the reference to A? Should it 404, should it show the original? 

tl;dr - we don't yet cater for *all* possible use cases. It should be fine to add a new setting that would control whether new translations should stay as draft, be always published, or follow the source page. Happy to review a PR for it

**Reply by stevejalim** [reply]
@zerolab We're in the use-case of needing to _not_ publish the new locale's page until the translations have landed, too.

I'm going to see if I can address this with a Signal handler, or failing that a fork of `wagtail-localize` with a new setting `WAGTAIL_LOCALIZE_IMMEDIATELY_PUBLISH_NEW_TRANSLATED_PAGES` (or perhaps something slightly less verbose!)

**Reply by krieghan** [reply]
My understanding is that there's a setting WAGTAILLOCALIZE_SYNC_LIVE_STATUS_ON_TRANSLATE.  Try setting it to False.

**Reply by thibaudcolas** [reply]
Just my two cents as a very casual wagtail-localize user for [guide.wagtail.org](https://guide.wagtail.org/) – I’ve also found this very surprising. In my mind if there’s a tradeoff here I’d have expected the default to be what scores the best on the principle of least astonishment (not sure if that’s "404" or "redirect to original").

Additionally form an accessibility perspective, having a translated page go live with the source language text will be problematic, as it means the page’s `lang` attribute will mismatch with the content (a failure of [WCAG SC 3.1.1 Language of Page](https://www.w3.org/WAI/WCAG22/quickref/?versions=2.1#language-of-page)).

**Reply by zerolab** [reply]
We did make this configurable via `WAGTAILLOCALIZE_SYNC_LIVE_STATUS_ON_TRANSLATE`: https://github.com/wagtail/wagtail-localize/blob/caf14b8e276a735f7536d3468762797af5c25315/wagtail_localize/operations.py#L82-L86 in https://github.com/wagtail/wagtail-localize/releases/tag/v1.4

It does sound that it may need to default to False. Feel free to submit a PR. It will require a breaking changes note in the changelog/release notes.

**Reply by stevejalim** [reply]
My reply to @krieghan got eaten somewhere, but yeah setting `WAGTAILLOCALIZE_SYNC_LIVE_STATUS_ON_TRANSLATE` to `True` gave me the desired behaviour. Thank you for the pointer @krieghan!

As for making it a default, I'm +1 on that.
